src/main/scala/streamer/CsrManager.scala

```diff
@@ -42,70 +40,65 @@ class CsrManager(
   // generate a vector of registers to store the csr state
   val csr = RegInit(VecInit(Seq.fill(csrNum)(0.U(32.W))))
 
-  // read and write csr cmd
+  // read write and start csr command
   val read_csr = io.csr_config_in.req.fire && !io.csr_config_in.req.bits.write
-  // streamerBusy2Idle has higher priority than the host request
-  val write_csr =
-    (io.csr_config_in.req.fire || io.streamerBusy2Idle) && io.csr_config_in.req.bits.write
-
-  // keep sending response to a read request until we receive the response ready signal
-  val keep_sending_csr_rsp = RegInit(0.B)
-  keep_sending_csr_rsp := io.csr_config_in.rsp.valid && !io.csr_config_in.rsp.ready
-
-  // a register to store the read request response data until the request is successful
-  val csr_rsp_data_reg = RegInit(0.U(32.W))
-
-  // store the csr data for later output because the address only valid when io.csr.fire
-  csr_rsp_data_reg := Mux(
-    read_csr,
-    csr(io.csr_config_in.req.bits.addr),
-    csr_rsp_data_reg
-  )
-
-  // streamer configuration valid signal
-  val config_valid = WireInit(0.B)
-
-  // check if the csr address overflow (access certain csr that doesn't exist)
-  def startCsrAddr = (csrNum - 1).U
+  val write_csr = io.csr_config_in.req.fire && io.csr_config_in.req.bits.write
+  val start_csr = io.csr_config_in.req.bits.write &&
+    (io.csr_config_in.req.bits.addr === (csrNum - 1).U) && io.csr_config_in.req.bits.data === 1.U
 
+  // assert the csr address is valid
   when(io.csr_config_in.req.fire) {
-
-    assert(
-      io.csr_config_in.req.bits.addr <= startCsrAddr,
-      "csr address overflow!"
-    )
-
+    assert(io.csr_config_in.req.bits.addr < csrNum.U, "csr address overflow!")
   }
 
-  // write req
+  // handle write req
   when(write_csr) {
     csr(io.csr_config_in.req.bits.addr) := io.csr_config_in.req.bits.data
   }
 
-  // handle read requests: keep sending response data until the request succeeds
+  // handle read requests: send the result directly. If the receiver
+  // is busy (ready not asserted), store the read result in a buffer.
+
+  // signal to indicate if a read request is in progress
+  val read_csr_busy = RegNext(
```

Comment:
This will introduce a can not reset bug in QeustaSim.  `read_csr_busy` will not be stable at the beginning. We need to explicitly reset it.